operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/action/batch/BatchCancelRequestAction.java

```diff
@@ -0,0 +1,432 @@
+import com.ibm.fhir.operation.bulkdata.model.type.JobParameter;
+import com.ibm.fhir.operation.bulkdata.util.BulkDataExportUtil;
+
+/**
+ * Per the specification, after a Bulk Data Request starts, the client may DELETE the request using the Content-Location (polling location)
+ *
+ * <code>
+ * DELETE [polling content location]
+ * Response - Success
+ * HTTP Status Code of 202 Accepted
+ * Optionally a FHIR OperationOutcome resource in the body
+ * Response - Error Status
+ * HTTP status code of 4XX or 5XX
+ * The body SHALL be a FHIR OperationOutcome resource in JSON format
+ * </code>
+ *
+ * @see http://hl7.org/fhir/uv/bulkdata/STU1/export/index.html#bulk-data-delete-request
+ */
+public class BatchCancelRequestAction implements BulkDataClientAction {
+    private static final String CLASSNAME = BatchCancelRequestAction.class.getName();
+    private static final Logger LOG = Logger.getLogger(CLASSNAME);
+
+    private static final BulkDataExportUtil export = new BulkDataExportUtil();
+
+    private static final List<String> INFLIGHT_STATUS = Arrays.asList("STARTING", "STARTED");
+
+    private CloseableHttpClient cli = null;
+    private String batchUrl = null;
+
+    private HttpClientContext ctx = null;
+
+    private Response.Status result = Response.Status.FORBIDDEN;
+
+    @Override
+    public void prepare(CloseableHttpClient cli, String batchUrl, String batchUser, String batchPass) {
+        this.cli = cli;
+
+        // Configure the local context.
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(AuthScope.ANY,
+                new UsernamePasswordCredentials(batchUser, batchPass));
+        ctx = HttpClientContext.create();
+        ctx.setCredentialsProvider(credsProvider);
+
+        this.batchUrl = batchUrl;
+    }
+
+    @Override
+    public Response.Status getResult() {
+        return result;
+    }
+
+    @Override
+    public void close() throws FHIROperationException {
+        if (cli != null) {
+            try {
+                cli.close();
+            } catch (IOException ex) {
+                LOG.throwing(CLASSNAME, "close", ex);
+                throw export.buildOperationException("Unable to close the Batch Client", IssueType.TRANSIENT);
+            }
+        }
+    }
+
+    @Override
+    public void run(String tenant, String job) throws FHIROperationException {
+        verifyJobExists(job);
+
+        List<JobExecutionResponse> jobExecutions = readJobExecutionsDetails(job);
+
+        verifyTenant(jobExecutions, tenant);
+
+        // Stop the Job's execution across all JobExecutions
+        stopJobExecutions(jobExecutions);
+
+        if (supportsDeleteJob()) {
+            // Delete The Job
+            deleteJob(job);
+        } else {
+            // The following is marked as not-found
+            throw export.buildOperationException("The Job is stopped intentionally with an in memory database.", IssueType.NOT_FOUND);
```

Comment:
please help me understand this one

Reply:
Chatted with Lee directly. 

The in Memory Persistence Impl that is used with JavaBatch does not support a the 'delete' (purgeJob) and always returns false which results in a 500. This is designed to signal to the user the reason and signal that it is deleted (when in fact it is just stopped).

Reply:
It'll only happen in this case when it's a non-production system.

Reply:
one idea I had is to change our default to use derby instead of this special in-memory config.  however, since neither one is recommended for production, i think its probably fine as-is.